improver/calibration/ensemble_calibration.py

```diff
@@ -625,8 +625,11 @@ def _create_temporal_coordinates(historic_forecasts):
         )
 
         # Create forecast period coordinate.
-        fp_point = np.unique(historic_forecasts.coord("forecast_period").points)
-        fp_coord = historic_forecasts.coord("forecast_period").copy(fp_point)
+        (fp_point,) = historic_forecasts.coord("forecast_period").points
+        fp_bounds = historic_forecasts.coord("forecast_period").bounds
+        fp_coord = historic_forecasts.coord("forecast_period").copy(
+            points=fp_point, bounds=fp_bounds
+        )
 
```

Comment:
Any reason not to just copy the forecast period coordinate as a whole?  Here as I see it, if the forecast period coordinate has more than one point, the coordinate copy statement will fail because the bounds are the wrong shape.  My understanding is the forecast period should only ever have one point anyway.

Reply:
You're quite right I can just copy the whole thing. I clearly got a bit blinkered. I have also added a check that it is single valued as I could not see this being checked anywhere; the alternative iris error is later on when trying to add a multi-valued coordinate without an associated dimension, but it could be a bit opaque.